lib/msal-browser/src/broker/BrokerClientApplication.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { version } from "../../package.json";
+import { BrokerAuthenticationResult, ServerTelemetryManager, AuthorizationCodeClient, BrokerAuthorizationCodeClient } from "@azure/msal-common";
+import { BrokerMessage } from "./BrokerMessage";
+import { BrokerMessageType, InteractionType } from "../utils/BrowserConstants";
+import { Configuration } from "../config/Configuration";
+import { BrokerHandshakeRequest } from "./BrokerHandshakeRequest";
+import { BrokerHandshakeResponse } from "./BrokerHandshakeResponse";
+import { BrokerAuthRequest } from "./BrokerAuthRequest";
+import { BrokerRedirectResponse } from "./BrokerRedirectResponse";
+import { RedirectRequest } from "../request/RedirectRequest";
+import { BrokerAuthResult } from "./BrokerAuthResult";
+import { ClientApplication } from "../app/ClientApplication";
+import { PopupRequest } from "../request/PopupRequest";
+
+/**
+ * Broker Application class to manage brokered requests.
+ */
+export class BrokerClientApplication extends ClientApplication {
+
+    constructor(configuration: Configuration) {
+        super(configuration);
+        this.logger.verbose("Acting as Broker");
+        this.listenForBrokerMessage();
+    }
+
+    /**
+     * 
+     */
+    private listenForBrokerMessage(): void {
+        window.addEventListener("message", this.handleBrokerMessage.bind(this));
+    }
+
+    /**
+     * 
+     * @param message 
+     */
+    private async handleBrokerMessage(message: MessageEvent): Promise<void> {
+        // Check that message is a BrokerHandshakeRequest
+        const clientMessage = BrokerMessage.validateMessage(message);
+        if (clientMessage) {
+            switch (clientMessage.data.messageType) {
+                case BrokerMessageType.HANDSHAKE_REQUEST:
+                    this.logger.verbose("Broker handshake request received");
+                    return await this.handleBrokerHandshake(clientMessage);
+                case BrokerMessageType.AUTH_REQUEST:
+                    this.logger.verbose("Broker auth request received");
+                    return await this.handleBrokerAuthRequest(clientMessage);
+                default:
+                    return;
+            }
+        }
+    }
+
+    /* eslint-disable */
```

Comment:
what are we disabling? Can you disable a specific rule not just the whole eslint?